Restore `main`, `module`, and `types`

The `exports` field is still fairly new for Typescript users, most of whom still have `"moduleResolution": "node"` in their `tsconfig.json`s. Such is the case of this repo as well.

The latest PR #404 from @benmccann removed the `main`, `module`, and `types` fields which node relies on for its module resolution.

```
src/blur/index.ts(1,29): error TS2307: Cannot find module 'imagetools-core' or its corresponding type declarations.
src/blur/index.ts(4,39): error TS2307: Cannot find module 'imagetools-core' or its corresponding type declarations.
src/blur/index.ts(42,33): error TS7006: Parameter 'config' implicitly has an 'any' type.
error Command failed with exit code 1.

```

AFAIK, the aforementioned fields can coexist with the `exports` fields.

--- a/.changeset/tiny-dryers-sneeze.md
+++ b/.changeset/tiny-dryers-sneeze.md
@@ -1,0 +1,7 @@
+---
+'imagetools-core': patch
+'rollup-plugin-imagetools': patch
+'vite-imagetools': patch
+---
+
+fix: restore package.json fields

--- a/.changeset/tiny-dryers-sneeze.md
+++ b/.changeset/tiny-dryers-sneeze.md
@@ -4,4 +4,4 @@
 'vite-imagetools': patch
 ---
 
-fix: restore package.json fields
+Fix type resolution by re-adding top-level types field

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -1,6 +1,9 @@
 {
   "name": "imagetools-core",
   "version": "3.2.2",
+  "main": "dist/index.cjs",
+  "module": "dist/index.mjs",
+  "types": "dist/index.d.ts",
   "exports": {
     "./package.json": "./package.json",
     ".": {

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -1,8 +1,6 @@
 {
   "name": "imagetools-core",
   "version": "3.2.2",
-  "main": "dist/index.cjs",
-  "module": "dist/index.mjs",
   "types": "dist/index.d.ts",
   "exports": {
     "./package.json": "./package.json",

--- a/packages/rollup/package.json
+++ b/packages/rollup/package.json
@@ -1,8 +1,6 @@
 {
   "name": "rollup-plugin-imagetools",
   "version": "3.0.11",
-  "main": "dist/index.cjs",
-  "module": "dist/index.mjs",
   "types": "dist/index.d.ts",
   "exports": {
     "./package.json": "./package.json",

--- a/packages/rollup/package.json
+++ b/packages/rollup/package.json
@@ -1,6 +1,9 @@
 {
   "name": "rollup-plugin-imagetools",
   "version": "3.0.11",
+  "main": "dist/index.cjs",
+  "module": "dist/index.mjs",
+  "types": "dist/index.d.ts",
   "exports": {
     "./package.json": "./package.json",
     ".": {

--- a/packages/vite/package.json
+++ b/packages/vite/package.json
@@ -2,6 +2,9 @@
   "name": "vite-imagetools",
   "description": "Load and transform images using a toolbox of import directives!",
   "version": "4.0.10",
+  "main": "dist/index.cjs",
+  "module": "dist/index.mjs",
+  "types": "dist/index.d.ts",
   "exports": {
     "./package.json": "./package.json",
     ".": {

--- a/packages/vite/package.json
+++ b/packages/vite/package.json
@@ -2,8 +2,6 @@
   "name": "vite-imagetools",
   "description": "Load and transform images using a toolbox of import directives!",
   "version": "4.0.10",
-  "main": "dist/index.cjs",
-  "module": "dist/index.mjs",
   "types": "dist/index.d.ts",
   "exports": {
     "./package.json": "./package.json",

